autoapi/mappers/base.py

```diff
@@ -322,9 +322,24 @@ def output_rst(self, root, source_suffix):
             detail_dir = obj.include_dir(root=root)
             ensuredir(detail_dir)
             path = os.path.join(detail_dir, f"index{source_suffix}")
+
+            # -- DEBUGGING --
             with open(path, "wb+") as detail_file:
                 detail_file.write(rst.encode("utf-8"))
 
+                if self.app.config.autoapi_render_in_single_page:
```

Comment:
This code is just a copy paste of the logic rendering each package or module. It basically iterates over its children and generates a page for them.

This only applies if the type of the children is declared in the list of "autoapi_render_in_single_page".